### DISCUSSION TITLE
[@types/jsonwebtoken] Wrong return - Sign method

### DISCUSSION THREAD

**Discussion by Fabio-Vicente** [DefinitelyTyped/DefinitelyTyped#61434] [Issues with a @types package]
Please verify the **sign** mehod on package "jsonwebtoken":

- That should has a string return type (a token really) instead void, that is the way it is actualy implemented.

How is implemented:
`export function sign(
    payload: string | Buffer | object,
    secretOrPrivateKey: Secret,
    callback: SignCallback,
): void;
export function sign(
    payload: string | Buffer | object,
    secretOrPrivateKey: Secret,
    options: SignOptions,
    callback: SignCallback,
): void;`

How should be implemented:
`export function sign(
    payload: string | Buffer | object,
    secretOrPrivateKey: Secret,
    callback: SignCallback,
): string;
export function sign(
    payload: string | Buffer | object,
    secretOrPrivateKey: Secret,
    options: SignOptions,
    callback: SignCallback,
): string;`

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
IMO the async versions returns nothing, so the definition seems to be correct:
https://github.com/auth0/node-jsonwebtoken/blob/master/sign.js#L194-L205

**Reply by Fabio-Vicente** [reply]
Ok, then. But what about the sync statament? Like it can be found on documentation, this function must return a string (https://github.com/auth0/node-jsonwebtoken#jwtsignpayload-secretorprivatekey-options-callback):

> (Synchronous) Returns the JsonWebToken as string

**Reply by peterblazejewicz** [reply]
if one does not provide callback, type definition return string:
```ts
// $ExpectType string
token = jwt.sign(testObject, "shhhhh", { keyid: "theKeyId" });
```
We can update tests with proper assertions. Is there an error you've got? Or just seeing inconsistency on project readme documentation

**Reply by Fabio-Vicente** [reply]
I've got a type error message provided by VSCode when trying to mock this method. Checking library, I've found the code showed above. Inside it, the return type is void, not string (a token, in the case).
